egui_demo_lib/src/apps/demo/plot_demo.rs

```diff
@@ -309,6 +311,117 @@ impl Widget for &mut LegendDemo {
     }
 }
 
+#[derive(PartialEq, Default)]
+struct CustomAxisDemo {}
+
+impl CustomAxisDemo {
+    fn logistic_fn() -> Line {
+        let values = Values::from_explicit_callback(
+            move |x| 1.0 / (1.0 + (-2.5 * (x - 2.0)).exp()),
+            0.0..5.0,
+            100,
+        );
+        Line::new(values)
+    }
+
+    #[allow(clippy::needless_pass_by_value)]
+    fn x_grid(input: GridInput) -> Vec<GridMark> {
+        // Note: this always fills all possible marks. For optimization, `input.bounds`
+        // could be used to decide when the low-interval grids (minutes) should be added.
+
+        let mut marks = vec![];
+
+        let steps_per_day = 24 * 12;
+        let fsteps_per_day = steps_per_day as f64;
+
+        let (min, max) = input.bounds;
+        let min = (min * fsteps_per_day).floor() as i32;
+        let max = (max * fsteps_per_day).ceil() as i32;
+
+        for i in min..=max {
+            let step_size = if i % steps_per_day == 0 {
+                // 1 day
+                1.0
+            } else if i % 12 == 0 {
+                // 1 hour
+                1.0 / 24.0
+            } else {
+                // 5min
+                1.0 / fsteps_per_day
+            };
+
+            let value = i as f64 / fsteps_per_day;
+
+            marks.push(GridMark { value, step_size });
+        }
+
+        marks
+    }
+}
+
+impl Widget for &mut CustomAxisDemo {
+    fn ui(self, ui: &mut Ui) -> Response {
+        // Make sure floor() still rounds down, even in presence of arithmetic imprecision
+        const EPS: f64 = 1e-6;
+
+        fn get_day(x: f64) -> f64 {
+            x.floor()
+        }
+        fn get_hour(x: f64) -> f64 {
+            (24.0 * x.fract() + EPS).floor()
+        }
+        fn get_minute(x: f64) -> f64 {
+            ((24.0 * x).fract() * 60.0 + EPS).floor()
+        }
+        fn get_percent(y: f64) -> f64 {
+            (100.0 * y + EPS).floor()
+        }
+
+        let x_fmt = |x, _range: &RangeInclusive<f64>| {
+            if x < 0.0 || x >= 5.0 {
+                // No labels outside value bounds
+                String::new()
+            } else if is_approx_integer(x) {
+                // Days
+                format!("Day {}", get_day(x))
+            } else {
+                // Hours and minutes
+                format!("{h}:{m:02}", h = get_hour(x), m = get_minute(x))
+            }
+        };
+
+        let y_fmt = |y, _range: &RangeInclusive<f64>| {
+            // Display only integer percentages
+            if !is_approx_zero(y) && is_approx_integer(100.0 * y) {
+                format!("{}%", get_percent(y))
```

Comment:
```suggestion
                format!("{:.0}%", (100.0 * y).round())
```

Reply:
`get_percent()` is used twice, not sure if the duplication would make things easier/clearer. Also, pre-formatting in two places would once just produce an intermediate string which is again formatted.

I left the method as-is for now, but open for suggestions:
```rs
fn get_percent(y: f64) -> f64 {
    (100.0 * y + EPS).floor()
}
```

Reply:
Ooh, I missed it was used twice.

`.round()` removed the need for `EPS`, and is more clear (imho), but it is a minor nit.

Reply:
Sure! Simplified it:
```rs
fn get_percent(y: f64) -> f64 {
    (100.0 * y).round()
}
```